packages/graphql/src/schema/augment/add-relationship-array-filters.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Directive, InputTypeComposer } from "graphql-compose";
+import pluralize from "pluralize";
+
+export function addRelationshipArrayFilters({
+    whereInput,
+    fieldName,
+    sourceName,
+    whereType,
+    directives,
+}: {
+    whereInput: InputTypeComposer<any>;
+    fieldName: string;
+    sourceName: string;
+    whereType: InputTypeComposer<any> | string;
+    directives: Directive[];
+}) {
+    whereInput.addFields(
+        (["ALL", "NONE", "SINGLE", "SOME"] as const).reduce(
+            (acc, filter) => ({
+                ...acc,
+                [`${fieldName}_${filter}`]: {
+                    type: whereType,
+                    directives: directives,
+                    // e.g. "Return Movies where all of the related Actors match this filter"
+                    description: `Return ${pluralize(sourceName)} where ${
+                        filter !== "SINGLE" ? filter.toLowerCase() : "one"
+                    } of the related ${pluralize(fieldName)} match this filter`,
```

Comment:
Rather than the field name, this used to be the related type name, which I think makes more sense for this description. You can see what I mean in the schema tests.

Reply:
I actually changed this because I thought it made more sense this way. I saw an example with the following type def:

```gql
type Movie {
    actors: [Person!]! @relationship(...)
    director: Person! @relationship(...)
}
```

Here `people` wouldn't make sense because it could be talking about the actors of the director. I do agree there are examples the other way round that make less sense too. I went this way because it was more specific so there's less room for confusion

Reply:
I'm happy to change it back if you think it was better before